### ISSUE TITLE
Integer decision variables?

### ISSUE THREAD

**Issue by mohamed82008** [robertfeldt/BlackBoxOptim.jl#108]
Does BlackBoxOptim support integer variants of the algorithms or only for real numbers? If not, any idea how easy or difficult would it be to add it? This is my first time using BlackBoxOptim but I am willing to add integer support if it won't take too much work since I have limited time. If this is not the appropriate channel for asking such a question, please let me know where to move to.

Thanks for the amazing package!

**Comment by robertfeldt** [comment]
Hi, thanks for the feedback and proposal!

We don't support it well today but you can try if basically wrapping the optimization in a round.(Int, v) before calling your fitness function (requiring int vector), i.e. something like:

```
f(v) = your_fitness_for_int_vector(round.(Int, v))
```
and then optimize f instead of `your_fitness_for_int_vector`. This might be problematic since the fitness landscape is flattened but for simpler and/or low-dimensional problems it might still be good enough.

Longer-term we definitely want to add mutation operators that work on Int vectors. But there might be changes needed throughout the package to really make this work; simply adding mutation operators for vectors of Int's might not be enough. Also, several of the optimization algorithms are built/designed for optimizing floating-point vectors so not clear they work well for int vectors.

So to sum up, my original intent was to support both int and floating point optimization but I'm not sure it is easy to do it well without potentially rethinking parts of the design. I'd be willing to explore it though.

**Comment by mohamed82008** [comment]
Thanks for your response. I used the rounding wrapper for now. I will read the package in whatever time I have available and see how I can contribute.
